### PR TITLE
fixing issue #11

### LINKED ISSUE #11: Standarize API to make pipelines "static" like
current use is:
var pipeMinify = require ('pipeline-minify-css')();

proposed use is:
var pipeMinify = require ('pipeline-minify-css');

resolves https://github.com/kenzanlabs/keystone/issues/7

### PR DESCRIPTION
**Pull request by cxoltero**
kenzanlabs/keystone#7
resolves #11

### CODE REVIEW

**Comment by obuckley-kenzan** [comment]
Can one of the admins verify this patch?

**Comment by obuckley-kenzan** [comment]
ok to test

File: src/index.js [hunk 1]
```diff
@@ -16,30 +16,33 @@ var config = {
...
-        return gulpIf(config.addSourceMaps, sourcemaps.write('maps'));
-      });
+module.exports = {
+  minifyCSS: function(options) {
+    if (options) {
+      handyman.log('Merging custom configuration');
+      config = handyman.mergeConf(config, options);
+    }
+
+    return minifyCSS();
```
Comment (obuckley-kenzan) [inline src/index.js]: could we just do `return makePipe()` here?

Comment (paul-barry-kenzan) [inline src/index.js]: @cxoltero @obuckley-kenzan 
I have to agree with Owen here. The `makePipe()` should be sufficient.

**Comment by cxoltero** [comment]
in this case, we could get rid of that part of the code, but then we will have to call it as follows:
minifyPipeline.minifyCSS()().
or either return it as:
return makePipe()();

**Comment by obuckley-kenzan** [comment]
Hmm, I see what you mean now with the double function calls.  

```
var pipeline = makePipe();

return pipeline();
```

**Comment by paul-barry-kenzan** [comment]
It's still under the hood though, so it won't affect the end user. Between an extra wrapping function or a simple `()()`, I prefer the `return makePipe()();` over the wrapping function.

**Comment by obuckley-kenzan** [comment]
How about something like...

```javascript
module.exports = {
  minifyCSS: function(options) {
    if (options) {
      handyman.log('Merging custom configuration');
      config = handyman.mergeConf(config, options);
    }

    return pipelineFactory(config);
  }
};

function pipelineFactory(config) {
  var pipeline =  lazypipe()
    .pipe(function() {
      return gulpIf(config.addSourceMaps, sourcemaps.init());
    })
    .pipe(minCSS, config.plugins.cleanCss)
    .pipe(function() {
      return gulpIf(config.concat, concat(config.concatFilename));
    })
    .pipe(function() {
      return gulpIf(config.addSourceMaps, sourcemaps.write('maps'));
    });

   return pipeline();
}
```

**Comment by cxoltero** [comment]
I think like @obuckley-kenzan's solution a little bit better than the ()() call.

**Comment by obuckley-kenzan** [comment]
edit: updated to encapsulate all business logic in the factory.  Thoughts?

@cxoltero / @paul-barry-kenzan ?

**Comment by cxoltero** [comment]
I agree with new logic, if approved, will apply to the other pipelines that are using the same structure.

**Comment by obuckley-kenzan** [comment]
@paul-barry-kenzan / @csharon thoughts?

**Comment by paul-barry-kenzan** [comment]
@cxoltero @obuckley-kenzan 
I like it.

**Comment by obuckley-kenzan** [comment]
wait for it.... what if we called the function `weldPipeline`  :D

**Comment by obuckley-kenzan** [comment]
This looks great @cxoltero , I'm really liking the `pipelineFactory` pattern as I start seeing them across our pipelines.  

Please update the README and this is good to me

**Pull request merged**